OpenRA.Mods.Common/Traits/Render/WithAttackAnimation.cs

```diff
@@ -53,12 +54,20 @@ public WithAttackAnimation(ActorInitializer init, WithAttackAnimationInfo info)
 			armament = init.Self.TraitsImplementing<Armament>()
 				.Single(a => a.Info.Name == info.Armament);
 			wsb = init.Self.Trait<WithSpriteBody>();
+
+			if (init.Self.TraitOrDefault<AttackTurreted>() != null)
```

Comment:
What if there is a turret which should not be the attacking thing? Then you will have a SpriteTurret, but its not the one who should attack. Thats why i implemented the AttackTurreted. Example: Vehicle with side-mounted machinegun, but a rotating radar on top as a turret.